mlir/test/Dialect/Index/index-canonicalize.mlir

```diff
@@ -556,3 +556,11 @@ func.func @sub_identity(%arg0: index) -> index {
   // CHECK-NEXT: return %arg0
   return %0 : index
 }
+
+// CHECK-LABEL: @castu_to_index
+func.func @castu_to_index() -> index {
+  // CHECK: index.constant 8000000000000
```

Comment:
Could you add the opposite direction here? That's where most of the logic is right?

Reply:
I'm trying to figure out the best way to test the other direction besides a test pass or unit test, because it's not possible to test the other direction via the canonicalizer -- the index dialect doesn't have a constant materializer for fixed-width integer constants...

Reply:
I gave up and just added unit tests